ml/backend/ggml/ggml.go

```diff
@@ -219,31 +235,23 @@ func (b *Backend) NewContext() ml.Context {
 	})
 
 	backends := make([]*C.struct_ggml_backend, len(b.gpus)+len(b.cpus))
```

Comment:
There's a bit of duplicate backend analyzing logic now between New() and NewContext() do we need it in both places or can it be shared somewhere?

Reply:
I know there's duplication for now but the way that the CPU/GPU backends are handled will be changing very soon once we handle splitting between them. The existing code is partially set up to handle this kind of split but it's not really enabled yet. It didn't seem to make much sense to restructure things to simply them when the other pending changes are likely to go in the opposite direction. As a result, I just kept things with the original structure for now.